packages/teams-js/src/public/presence.ts

```diff
@@ -0,0 +1,226 @@
+import { callFunctionInHostAndHandleResponse } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { ResponseHandler } from '../internal/responseHandler';
+import { ApiName, ApiVersionNumber, getApiVersionTag } from '../internal/telemetry';
+import { errorNotSupportedOnPlatform, FrameContexts } from './constants';
+import { ErrorCode } from './interfaces';
+import { runtime } from './runtime';
+import { ISerializable } from './serializable.interface';
+
+/**
+ * v2 APIs telemetry file: All of APIs in this capability file should send out API version v2 ONLY
+ */
+const presenceTelemetryVersionNumber: ApiVersionNumber = ApiVersionNumber.V_2;
+
+/**
+ * Represents a user's presence status
+ */
+export enum PresenceStatus {
+  /**
+   * User is available and can be contacted
+   */
+  Available = 'Available',
+
+  /**
+   * User is busy and should not be disturbed
+   */
+  Busy = 'Busy',
+
+  /**
+   * User has explicitly set their status to "Do Not Disturb" and should not be contacted
+   */
+  DoNotDisturb = 'DoNotDisturb',
+
+  /**
+   * User is temporarily away from their device
+   */
+  Away = 'Away',
+
+  /**
+   * User is offline and cannot be contacted
+   */
+  Offline = 'Offline',
+}
+
+/**
+ * Represents a user's presence information
+ */
+export interface UserPresence {
+  /**
+   * User's current presence status
+   */
+  status: PresenceStatus;
+
+  /**
+   * Optional custom status message
+   * @since 2.0.0
+   */
+  customMessage?: string;
+}
+
+/**
+ * Parameters for getting a user's presence
+ */
+export interface GetPresenceParams {
+  /**
+   * The user's UPN (email) to get presence for
+   */
+  upn: string;
+}
+
+/**
+ * Parameters for setting presence
+ */
+export interface SetPresenceParams {
+  /**
+   * New presence status to set
+   */
+  status: PresenceStatus;
+
+  /**
+   * Optional custom status message
+   * @since 2.0.0
+   */
+  customMessage?: string;
+}
+
+/**
+ * Response handler for presence information
+ */
+class UserPresenceResponseHandler extends ResponseHandler<UserPresence, UserPresence> {
+  public validate(response: UserPresence): boolean {
+    return response !== undefined && Object.values(PresenceStatus).includes(response.status);
+  }
+
+  public deserialize(response: UserPresence): UserPresence {
+    return response;
+  }
+}
+
+/**
+ * Response handler for void responses
+ */
+class VoidResponseHandler extends ResponseHandler<void, void> {
+  public validate(response: void): boolean {
+    return response === undefined;
+  }
+
+  public deserialize(response: void): void {
+    return response;
+  }
+}
+
+/**
+ * Serializable wrapper for GetPresenceParams
+ */
+class SerializableGetPresenceParams implements ISerializable {
```

Comment:
This is needed and correct because you are passing a complex interface (as opposed to a primitive or array of primitives). If you instead decided to pass the contents of GetPresenceParams as an array of primitives you wouldn't need to make a class like this. Both approaches are/can be correct